lib/datadog/grok/src/parse_grok.rs

```diff
@@ -786,4 +791,37 @@ mod tests {
             ),
         ]);
     }
+
+    #[test]
+    fn parses_with_new_lines() {
+        test_full_grok(vec![
+            (
+                "%{data:field}",
+                "a\nb",
+                Ok(Value::from(btreemap! {
+                    "field" => "a\nb"
+                })),
+            ),
+            (
+                "%{data:line1}\n%{data:line2}",
+                "a\nb",
+                Ok(Value::from(btreemap! {
+                    "line1" => "a",
+                    "line2" => "b"
+                })),
+            ),
+            // disable DOTALL mode
+            ("(?-m)%{data:field}", "a\nb", Err(Error::NoMatch)),
+            // (?s) is not supported by the underlying regex engine(onig) - it uses (?m) instead, so we convert it silently
+            (
+                "(?s)%{data:field}",
+                "a\nb",
+                Ok(Value::from(btreemap! {
+                    "field" => "a\nb"
+                })),
+            ),
+            // disable DOTALL mode with (?-s)
+            ("(?-s)%{data:field}", "a\nb", Err(Error::NoMatch)),
```

Comment:
I think a test that might be useful is a pattern that changes the mode several times, which I presume is supported?

Something like:

`"(?-s)%{data:field} (?s)%{data::field}", "abc d\ne"`

Reply:
Good idea, add this test case.